### PR TITLE
Adds replaceUser route and remove replaceIfExists option in createUser

### PR DESCRIPTION
**Pull request by samniisan**
PR has been changed to remove to remove the replaceIfExists option.
fixes: https://github.com/kuzzleio/kuzzle-sdk/issues/21

### CODE REVIEW

**Review by stafyniaksacha — DISMISSED** [review]

**Review by benoitvidis — DISMISSED** [review]

**Review by dbengsch — DISMISSED** [review]

File: src/Security/Security.php [hunk 1]
```diff
@@ -123,9 +123,48 @@ public function createUser($id, array $content, array $options = [])
         ];
 
         if (array_key_exists('replaceIfExist', $options)) {
-            $action = 'createOrReplaceUser';
+            $fetchResponse = $this->kuzzle->query(
```
Comment (ballinette) [inline src/Security/Security.php]: Suggestion for a cleaner code: replace lines 125-147 by following:

```php
if ($options['replaceIfExist']) {
  $fetchResponse = $this->kuzzle->query(
    $this->buildQueryArgs('getUser'),
    ['_id' => $id]
  );

  if ($fetchResponse["error"] === null && $fetchResponse["result"]["_id"] === $id) {
    $action = 'replaceUser';
  }
}

$response = $this->kuzzle->query(
  $this->buildQueryArgs($action),
  $data,
  $options
);
```

**Review by ballinette — DISMISSED** [review]

**Comment by dbengsch** [comment]
Blocking: ReplaceIfExists Does not match with the new auth refactor

**Review by ballinette — APPROVED** [review]

**Comment by ballinette** [comment]
hmmm. I validated, but... why do we remove the `replaceIfExists` option ? it should be consistent with other SDKs...

I guess it should be implemented as for Javascript SDK : https://github.com/kuzzleio/sdk-javascript/pull/208

**Comment by dbengsch** [comment]
We are removing it from all SDKs.

**Review by xbill82 — APPROVED** [review]

**Pull request merged**